### ISSUE TITLE
codebase cannot be None when using GerritChangeSource

### ISSUE THREAD

**Issue by LKDbigpig214** [buildbot/buildbot#5400]
Traceback (most recent call last):
	  File "/home/goodix/workspace/buildbot/master/sandbox/lib/python3.5/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
	    result = g.send(result)
	  File "/home/goodix/workspace/buildbot/master/sandbox/lib/python3.5/site-packages/buildbot/changes/gerritchangesource.py", line 177, in addChange
	    self.master.db.sourcestamps.findOrCreateId(**stampdict))
	  File "/home/goodix/workspace/buildbot/master/sandbox/lib/python3.5/site-packages/twisted/internet/defer.py", line 1613, in unwindGenerator
	    return _cancellableInlineCallbacks(gen)
	  File "/home/goodix/workspace/buildbot/master/sandbox/lib/python3.5/site-packages/twisted/internet/defer.py", line 1529, in _cancellableInlineCallbacks
	    _inlineCallbacks(None, g, status)
	--- <exception caught here> ---
	  File "/home/goodix/workspace/buildbot/master/sandbox/lib/python3.5/site-packages/buildbot/changes/gerritchangesource.py", line 332, in outLineReceived
	    yield self.change_source.lineReceived(line)
	  File "/home/goodix/workspace/buildbot/master/sandbox/lib/python3.5/site-packages/buildbot/changes/gerritchangesource.py", line 249, in addChangeFromEvent
	    'properties': properties})
	  File "/home/goodix/workspace/buildbot/master/sandbox/lib/python3.5/site-packages/buildbot/changes/gerritchangesource.py", line 177, in addChange
	    self.master.db.sourcestamps.findOrCreateId(**stampdict))
	  File "/home/goodix/workspace/buildbot/master/sandbox/lib/python3.5/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
	    result = g.send(result)
	  File "/home/goodix/workspace/buildbot/master/sandbox/lib/python3.5/site-packages/buildbot/db/sourcestamps.py", line 58, in findOrCreateId
	    assert codebase is not None, "codebase cannot be None"
	builtins.AssertionError: codebase cannot be None

then I find addChange in changes/gerritchangesource.py, the code is:
   ```
stampdict = {
            "branch": chdict["branch"],
            "revision": chdict["revision"],
            "patch_author": chdict["author"],
            "patch_comment": chdict["comments"],
            "repository": chdict["repository"],
            "project": chdict["project"],
        }

        stampid, found_existing = yield(
             self.master.db.sourcestamps.findOrCreateId(**stampdict))
```

There is no codebase in stampdict, but in db/sourcestamps.findOrCreateId, the code is:
 ```
 def findOrCreateId(self, branch=None, revision=None, repository=None,
                       project=None, codebase=None, patch_body=None,
                       patch_level=None, patch_author=None,
                       patch_comment=None, patch_subdir=None):
        tbl = self.db.model.sourcestamps

        assert codebase is not None, "codebase cannot be None"
```

**Comment by spenserlee** [comment]
I think this is a duplicate of #5190.

@0x8000-0000 posted a potenial workaround (not yet merged):

> Adding line 177 in buildbot/changes/gerritchangesource.py seems to work around the problem.
> 
>     168 @defer.inlineCallbacks
>     169 def addChange(self, chdict):
>     170 stampdict = {
>     171 "branch": chdict["branch"],
>     172 "revision": chdict["revision"],
>     173 "patch_author": chdict["author"],
>     174 "patch_comment": chdict["comments"],
>     175 "repository": chdict["repository"],
>     176 "project": chdict["project"],
>     177 "codebase": '',
>     178 }

**Comment by p12tic** [comment]
I think this has been fixed in #5219 and released in Buildbot 2.9.0. Please reopen the issue if that's not the case.
